### ISSUE TITLE
Tasks overlap Timeslots

### ISSUE THREAD

**Issue by rejkid** [JamesBremner/Agents2Tasks#8]
Implementation of a new algorithm where tasks do overlap timeslots.

Assigning Agent A to a task with the name T prevents her from being assigned to any other task named T for today and tomorrow

**Comment by JamesBremner** [comment]
When timeslots are independent, it does not matter what they are named.  You can call them whatever you like, just as long as they are unique strings.

Now that we need to know whether or not two timeslots are in the same day, the name must be parsed to extract the data.  So the  date format must be specified.

Currently, you are using this:

![image](https://github.com/JamesBremner/Agents2Tasks/assets/2046227/a496a7cc-c72b-41f5-ae7b-b641689d4563)

This _could_ be parsed.  However, I would like something that is easier to work with.  I suggest

```
202310220820
202310221000
```

This uses a 24 hour clock for the time of day..

Acceptable?

**Comment by JamesBremner** [comment]
> agent can't be assigned to the same task twice a day.

This, or perhaps

agents cannot be assigned to more than one task per day

**Comment by rejkid** [comment]
James - 
1) In my case agents can be assign to different tasks per day (but not the same) - so John can be assigned to both Acolyte and Cleaner in the same day, but he can not be assigned to Acolyte again on the same day
2) The date format is absolutely acceptable

**Comment by JamesBremner** [comment]
You sometimes have multiple tasks with the same name in one day.

t 28/OCT/2023/8:30 teacher teacher cleaner 

Are the two "teacher" tasks both the same task has all teachers in previous slots?

**Comment by rejkid** [comment]
Sorry James can you rephrase last question please - I don't quite understand it?

**Comment by JamesBremner** [comment]
First time slot

t 28/OCT/2023/8:30 teacher teacher cleaner

Agent Alice is assigned to the first teacher task,  agent Bob to the second

Second time slot

t 28/OCT/2023/10:30 teacher teacher cleaner

Are both Alice and Bob unable to do both teacher jobs?  Or can they swap the jobs with each other and still not have been assigned to the "same" job in the same day.

Best guess: Agent can only be assigned to one task per day with the same name.

**Comment by rejkid** [comment]
That is correct. 
Also can we rephrase the "Agent can only be assigned to the same task once a day" to "Agent can only be assigned to the same task once per two days" - this will cover the case where someone is assigned task on the weekend twice Saturday and Sunday (which should not happen).

**Comment by JamesBremner** [comment]
> That is correct.

Which option is correct?  The one where assigning Alice to a task with the name "teacher" prevents her from being assigned to any other task named "teacher" for today and tomorrow?

**Comment by rejkid** [comment]
I am sorry. I know. It's just that is is a schedule for the church where agent can't be assigned to the task twice per weekend.

**Comment by rejkid** [comment]
Assigning Alice to a task with the name "teacher" prevents her from being assigned to any other task named "teacher" for today and tomorrow - that is correct

**Comment by JamesBremner** [comment]
```
a Alice 1 teacher 

t 202311091000 teacher 
t 202311091300 teacher 
t 202311101000 teacher 
t 202311111000 teacher 

A 202311091000 Alice to teacher
A 202311111000 Alice to teacher
```

**Comment by rejkid** [comment]
James, I have run into another problem. This is not a2t problem but let me explain what the issue is. a2t is an external console application that reads the data input file and writes an output file with the result. The problem is that if I call a2t console app from c# it always outputs an empty file (I guess some security issue). If I do this process manually by executing the a2t console app by myself (passing input and output files as arguments) it creates non empty output result file as expected. People on internet suggested to call into Dll if I have access to it. Is it possible to compile a2t as dll with one function to call? If not then don't worry about it I will have to come up with some other solution.

**Comment by JamesBremner** [comment]
> , I have run into another problem.

Please do not add irrelevant posts to an issue.  Different problems should each get their own issue.  Piling different problems onto the end of existing issues makes for a confusing mess.

I realize that it is not always easy to tell what is relevant and and what is not. I am happy to open new issues for cients when they make an honest mistake.  However, if you find yourself typing 'another problem' or anything similar then you almost certainly need to open a new issue.

**Comment by rejkid** [comment]
Attached is the result file. The test passed
[a2t_result.txt](https://github.com/JamesBremner/Agents2Tasks/files/13321148/a2t_result.txt)
